### DISCUSSION TITLE
Add dotenv and polyfills globally

### DISCUSSION THREAD

**Discussion by nanotime** [brunch/brunch#1869] [General]
Well after a lot of search and tests, i can't make `dotenv` works, or polyfills like `whatwg-fetch` in my environment.

Hmm, i'm complete noob with this kind of tools, i've been using things like `create-react-app` so, yep, this is new for me.

I tried to use the order of compiling, and `entryFiles` to bootstrap this kind of things but nothing works, and can't figure some solution in the docs or google... here is my actual configuration:

```javascript
exports.files = {
  javascripts: {
    joinTo: {
      'dependencies.js': /^(?!app\/)/,
      'bundle.js': /^app\//
    },
    entryPoints: {
      'app/confs.js': 'bundle.js'
    }
  },
  stylesheets: {
    joinTo: 'app.css'
  },
};

exports.plugins = {
  babel: { presets: ['latest', 'react'] }
};
```

**confs.js**

```javascript
require('dotenv').config();
require('whatwg-fetch');
```

Any recomendation? Hint? The idea is just to get the `dotenv` capabilities in all of my files, and add support for fetch with the polifyll.

**Comment by shvaikalesh** [comment]
Hey @nanotime, we appreciate the issue.

`joinTo` and `entryPoints` can't be used together. I am sorry that docs wasn't enough clear on this.
We recommend using `entryPoints`:

```js
entryPoints: {
  'app/shims.js': 'shims.js',
  'app/init.js': 'app.js',
}
```

**Comment by nanotime** [comment]
@shvaikalesh ok, i'll try that.

One question: ¿Why you recommend `entryPoints` over `joinTo`?

**Comment by denysdovhan** [comment]
@nanotime `entryPoints` are a bit smarter than just `joinTo`. Here is a note from docs:

> Only the things you require will be included into an entryPoint bundle.
>This means non-app/non-npm dependencie won't be included. Also means only statically analyzable requires will work.
